lib/parse.js

```diff
@@ -35,7 +40,24 @@ module.exports = function(code, options) {
     },
   };
 
+  if (options.requireConfigFile !== false) {
+    const config = loadPartialConfig(opts);
+
+    if (config !== null) {
+      if (!config.hasFilesystemConfig()) {
+        throw new Error(
+          `No Babel config file detected for ${
+            config.options.filename
+          }. Either disable config file checking with requireConfigFile: false, or configure Babel so that it can find the config files.`
+        );
+      }
+
+      opts = config.options;
+    }
+  }
+
   let ast;
+
   try {
     ast = parse(code, opts);
```

Comment:
Not related to this PR, but do we want to throw a helpful error if this ends up returning `null`? Otherwise I assume it will throw if someone tries to run ESLint on a file that is in the user's `.babelrc`?

Reply:
Question about this: It looks like `babelCore#parseSync` returns `null` when there isn't a config, but since we have a default configuration we pass through if a file isn't found with `requireConfig: false` and it warns when a config isn't found with `requireConfig: true`, I'm not seeing how `babel-eslint` can get into this state.